dev/tools/localization/gen_l10n_templates.dart

```diff
@@ -205,15 +187,48 @@ const String baseClassMethodTemplate = '''
   String @(name)(@(parameters));
 ''';
 
+// DELEGATE CLASS TEMPLATES
+
+const String delegateClassTemplate = '''
+class _@(class)Delegate extends LocalizationsDelegate<@(class)> {
+  const _@(class)Delegate();
+
+  @override
+  Future<@(class)> load(Locale locale) {
+    @(loadBody)
+  }
+
+  @override
+  bool isSupported(Locale locale) => <String>[@(supportedLanguageCodes)].contains(locale.languageCode);
+
+  @override
+  bool shouldReload(_@(class)Delegate old) => false;
+}
+
+@(lookupFunction)''';
+
+const String loadBodyTemplate = '''return SynchronousFuture<@(class)>(@(lookupName)(locale));''';
+
+const String loadBodyDeferredLoadingTemplate = '''return @(lookupName)(locale);''';
```

Comment:
In the event that we're using deferred loading, what happens for users of other platforms like mobile and desktop here? Does this become asynchronous, or does it mimic `SynchronousFuture`?

Reply:
If it does become truly asynchronous, we should definitely make note of the distinction in the tool's help text since this changes how the Flutter apps handle localizations for non-web platforms

Reply:
I changed it now, so that the other platforms are using `SynchronousFuture` as it is only on the web that we have support for deferred loading.